odk1-src/form-functions.rst

```diff
@@ -0,0 +1,719 @@
+    .. csv-table:: survey
+      :header: type, name, label, hint, constraint, constraint_message
+
+      select_multiple colors, color_prefs, What colors do you like?, Select three., count-selected(.)=3, Select exactly three.
+
+    .. csv-table:: choices
+      :header: list_name, name, label
+
+      colors, red, Red
+      colors, blue, Blue
+      colors, yellow, Yellow
+      colors, green, Green
+      colors, orange, Orange
+      colors, purple, Purple
+
+.. function:: jr:choice-name(choice_name, 'select_question')
+
+  Returns the label value, in the active language, associated with the :arg:`choice_name` in the list of choices for the :arg:`select_question`.
+  
+  .. note::
+  
+    You have to wrap the :arg:`select_question` reference in quotes.
+    
+    .. code-block:: none
+    
+      '${question_name}'
+      
+  .. container:: details  
+  
+    .. image:: /img/form-functions/choice-name-multi-lang-english-0.* 
+      :alt: A multi-select widget in Collect. The label is "What colors do you like?" Several color names are presented as options. Red, Green, and Purple are selected.
+
+    .. image:: /img/form-functions/choice-name-multi-lang-english-1.* 
+      :alt: A note widget in Collect. The label is "Selected colors". The hint text is "Red, Green, Purple."
+
+    .. image:: /img/form-functions/choice-name-multi-lang-spanish-0.* 
+      :alt: A multi-select widget in Collect. The label is "¿Qué colores te gustan?" Several color names, in Spanish, are presented as options. Rojo, Verde, and Púrpura are selected.
+
+    .. image:: /img/form-functions/choice-name-multi-lang-spanish-1.* 
+      :alt: A note widget in Collect. The label is "Colores seleccionados." The hint text is "Rojo, Verde, Púrpura".
+
+    .. rubric:: XLSForm
+
+    .. csv-table::  survey
+      :header: type, name, label::English, label::Español, hint::English, hint:Español, calculation
+
+      select_multiple colors, color_prefs, What colors do you like?, ¿Qué colores te gustan?, Select three., Seleccione tres.
+      calculate, color_0, , , , ,"jr:choice-name( selected-at(${color_prefs}, 0), '${color_prefs}')"
+      calculate, color_1, , , , ,"jr:choice-name( selected-at(${color_prefs}, 2), '${color_prefs}')"
+      calculate, color_2, , , , ,"jr:choice-name( selected-at(${color_prefs}, 2), '${color_prefs}')"
+      note, color_note, Selected colors:, Colores seleccionados:, ${color_0} <br> ${color_1} <br> ${color_2}, ${color_0} <br> ${color_1} <br> ${color_2}
+
+    .. csv-table:: choices
+      :header: list_name, name, label::English, label::Español
+
+      colors, red, Red, Rojo
+      colors, blue, Blue, Azul
+      colors, yellow, Yellow, Amarillo
+      colors, green, Green, Verde
+      colors, orange, Orange, Anaranjado
+      colors, purple, Purple, Púrpura
+
+.. _repeat-functions:
+    
+Repeat groups
+~~~~~~~~~~~~~~~~
+
+.. admonition:: Helpful terms
+
+  .. glossary::
+    :sorted:
+
+    nodeset
+
+      A collection of response values. 
```

Comment:
It may be worth mentioning that this is as in XPath -- a set of any XML nodes. You can build a nodeset a number of ways. For example, if you have nodes `/data/name` and `/data/age`, you can do `/data/name | /data/age` to build up a nodeset that contains both nodes. Functions like `randomize` return nodesets not necessarily of responses (generally of choices).

Reply:
Could one do:

``${question_name_1} | ${question_name_2}``

?

Reply:
> Functions like randomize return nodesets not necessarily of responses (generally of choices).